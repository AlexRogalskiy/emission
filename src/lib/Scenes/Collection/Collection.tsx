import { Box, Theme } from "@artsy/palette"
import { Collection_collection } from "__generated__/Collection_collection.graphql"
import React, { Component } from "react"
import { FlatList } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import { CollectionHeaderContainer as CollectionHeader } from "./Components/CollectionHeader"
import { CollectionFeaturedArtistsContainer as CollectionFeaturedArtists } from "./Components/FeaturedArtists"

interface CollectionProps {
  collection: Collection_collection
  artworks: any
}

interface CollectionState {
  sections: Array<{ type: string; data: any }>
}

export class Collection extends Component<CollectionProps, CollectionState> {
  state = {
    sections: [],
  }

  componentDidMount() {
    const sections = []

    sections.push({
      type: "featuredArtists",
      data: {
        artists: [],
      },
    })

    this.setState({
      sections,
    })
  }

  renderItem({ item }) {
    switch (item.type) {
      case "featuredArtists":
        return (
          <Box>
            <CollectionFeaturedArtists featuredArtists={this.props.collection} />
          </Box>
        )
      default:
        return null
    }
  }

  render() {
    const { sections } = this.state

    return (
      <Theme>
        <Box>
          <FlatList
            keyExtractor={(_item, index) => String(index)}
            data={sections}
            ListHeaderComponent={<CollectionHeader collection={this.props.collection} />}
            renderItem={item => (
              <Box px={2} pb={2}>
                {this.renderItem(item)}
              </Box>
            )}
          />
        </Box>
      </Theme>
    )
  }
}

export const CollectionContainer = createFragmentContainer(Collection, {
  collection: graphql`
    fragment Collection_collection on MarketingCollection {
      ...CollectionHeader_collection
      ...FeaturedArtists_featuredArtists
    }
  `,
})
