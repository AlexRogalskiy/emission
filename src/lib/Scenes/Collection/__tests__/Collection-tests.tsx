import { CollectionTestsQuery } from "__generated__/CollectionTestsQuery.graphql"
import React from "react"
import { graphql, QueryRenderer } from "react-relay"
import ReactTestRenderer from "react-test-renderer"
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
import { CollectionContainer, FilterArtworkButton, FilterArtworkButtonContainer } from "../Collection"

jest.unmock("react-relay")

describe("Collection", () => {
  let environment: ReturnType<typeof createMockEnvironment>
  const TestRenderer = () => (
    <QueryRenderer<CollectionTestsQuery>
      environment={environment}
      query={graphql`
        query CollectionTestsQuery @relay_test_operation {
          marketingCollection(slug: "doesn't matter") {
            ...Collection_collection
          }
        }
      `}
      variables={{ hello: true }}
      render={({ props, error }) => {
        if (props) {
          return <CollectionContainer collection={props.marketingCollection} />
        } else if (error) {
          console.log(error)
        }
      }}
    />
  )

  beforeEach(() => {
    environment = createMockEnvironment()
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it("renders a snapshot", () => {
    const renderer = ReactTestRenderer.create(<TestRenderer />)
    environment.mock.resolveMostRecentOperation(operation => {
      return MockPayloadGenerator.generate(operation)
    })
    expect(renderer.toJSON()).toMatchSnapshot()
  })

<<<<<<< Updated upstream
  it("does not display a filter artworks button by default", () => {
    const root = ReactTestRenderer.create(<TestRenderer />).root

    expect(root.findAllByType(FilterArtworkButtonContainer)).toHaveLength(0)
    expect(root.findAllByType(FilterArtworkButton)).toHaveLength(0)
  })

  /**  TODO: Pair with MX to complete get these assertions to pass
   * How do we mock an update to the state object and the Native Emission Module to
   * get these components to render
   */
  xit("does display a filter artworks button when artworks grid when artworks grid is in view", () => {
    // expect(root.findAllByType(FilterArtworkButtonContainer)).toHaveLength(1)
    // expect(root.findAllByType(FilterArtworkButton)).toHaveLength(1)
=======
  it("does not render modal filter button when feature flag disabled", () => {
    const renderer = ReactTestRenderer.create(<TestRenderer />)
    environment.mock.resolveMostRecentOperation(operation => {
      return MockPayloadGenerator.generate(operation)
    })
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it("does render modal filter button when feature flag enabled", () => {
    const renderer = ReactTestRenderer.create(<TestRenderer />)
    environment.mock.resolveMostRecentOperation(operation => {
      return MockPayloadGenerator.generate(operation)
    })
    expect(renderer.toJSON()).toMatchSnapshot()
>>>>>>> Stashed changes
  })
})
