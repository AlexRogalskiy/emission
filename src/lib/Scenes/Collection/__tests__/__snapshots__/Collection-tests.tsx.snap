// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Renders collection  correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <CollectionHeader
        collection={
          Object {
            "description": "<p>From <a href=\\"https://www.someartsyimage.net/artist/jean-michel-basquiat\\">Jean-Michel Basquiat</a> to <a href=\\"https://www.someartsyimage.net/artist/shepard-fairey\\">Shepard Fairey</a>, street artists are known for their creative expressions of rebellion.</p>",
            "headerImage": "http://imageuploadedbymarketingteam.jpg",
            "image": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    "imageUrl": "https://defaultmostmarketableartworkincollectionimage.jpg",
                  },
                },
              ],
            },
            "title": "Street Art Now",
          }
        }
      />
    }
    data={
      Array [
        Object {
          "data": Object {
            "artists": Array [],
          },
          "type": "featuredArtists",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          mb={2}
          style={
            Array [
              Object {
                "marginBottom": 20,
              },
            ]
          }
        >
          <AROpaqueImageView
            height={204}
            imageURL="http://imageuploadedbymarketingteam.jpg"
            width={750}
          />
        </View>
        <Text
          color="#000"
          fontFamily="ReactNativeAGaramondPro-Regular"
          fontSize="32px"
          lineHeight="38px"
          ml={2}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 32,
                "lineHeight": 38,
                "marginLeft": 20,
              },
              Object {},
            ]
          }
        >
          Street Art Now
        </Text>
        <Text
          color="#000"
          fontFamily="ReactNativeAGaramondPro-Regular"
          fontSize="18px"
          lineHeight="26px"
          mt={2}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 18,
                "lineHeight": 26,
                "marginTop": 20,
              },
              Object {},
            ]
          }
        >
          &lt;p&gt;From &lt;a href="https://www.someartsyimage.net/artist/jean-michel-basquiat"&gt;Jean-Michel Basquiat&lt;/a&gt; to &lt;a href="https://www.someartsyimage.net/artist/shepard-fairey"&gt;Shepard Fairey&lt;/a&gt;, street artists are known for their creative expressions of rebellion.&lt;/p&gt;
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          pb={2}
          px={2}
          style={
            Array [
              Object {
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
